Sources/UIApplication+handleSDLEvents.swift

```diff
@@ -143,3 +150,127 @@ extension SDL_Event {
         return TimeInterval(self.common.timestamp) / 1000
     }
 }
+
+extension UIEvent {
+    static func from(_ event: SDL_Event) -> UIEvent? {
+        switch SDL_EventType(event.type) {
+        case SDL_FINGERDOWN:
+            let newTouch = UITouch(
+                touchId: Int(event.tfinger.fingerId),
+                at: CGPoint(
+                    x: CGFloat(event.tfinger.x),
+                    y: CGFloat(event.tfinger.y)
+                ),
+                timestamp: event.timestampInSeconds
+            )
+
+            if
+                let firstExistingEvent = UIEvent.activeEvents.first,
+                let _ = firstExistingEvent.allTouches?.first(where: {$0.touchId == event.tfinger.fingerId})
+            {
+                // Found a matching event, adding current touch to it and returning
+                firstExistingEvent.allTouches?.insert(newTouch)
+                return firstExistingEvent
+            } else {
+                // No matching event found, creating a new one
+                return UIEvent(touch: newTouch)
```

Comment:
something about this implementation broke last I looked at it - which is why I started implementing proper multitouch.

this is half implementing multitouch because it pays attention to `touchId` - for this to work in a minimal implementation we need to remove that and always assume `touchId == 0`

Reply:
@ephemer do you remember what exactly broke? this current state seems totally fine to me, even when attempting to use multitouch. If I make the assumption of `touchId == 0` (e.g. replacing `event.tfinger.fingerId` with `0`) then the sheet goes crazy when using multiple fingers.